internal/commands/publish.go

```diff
@@ -374,35 +384,44 @@ func endOfSupportFor(publishDate time.Time) string {
 	return endOfNinthMonth.Format(publishDateFormat)
 }
 
-func (p Publish) attachLicenseFile(slug string, releaseID int, version *releaseVersion) (string, error) {
+func (p Publish) attachLicenseFiles(slug string, releaseID int, version *releaseVersion) ([]string, error) {
+	var attachedLicenseFileNames []string
+
 	if version.IsGA() {
 		productFiles, err := p.PivnetProductFilesService.List(slug)
 		if err != nil {
-			return "", err
+			return attachedLicenseFileNames, err
 		}
 
-		productFile, found := findMatchingOSL(productFiles, version)
+		licenseFiles, found := findMatchingLicenseFiles(productFiles, version)
 		if !found {
-			return "", errors.New("required license file doesn't exist on Pivnet")
+			return attachedLicenseFileNames, errors.New("required license file doesn't exist on Pivnet")
 		}
 
-		err = p.PivnetProductFilesService.AddToRelease(slug, releaseID, productFile.ID)
-		if err == nil {
-			return productFile.Name, nil
-		} else {
-			return "", err
+		for _, licenseFile := range licenseFiles {
+			err = p.PivnetProductFilesService.AddToRelease(slug, releaseID, licenseFile.ID)
+
+			if err == nil {
+				attachedLicenseFileNames = append(attachedLicenseFileNames, licenseFile.Name)
```

Comment:
The control flow block here can be written without an ["else"](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow).

```
for _, licenseFile := range licenseFiles {
  if err := p.PivnetProductFilesService.AddToRelease(slug, releaseID, licenseFile.ID); err != nil {
    return attachedLicenseFileNames, err
  }
  attachedLicenseFileNames = append(attachedLicenseFileNames, licenseFile.Name)
}

Reply:
Noted and changed!  i've never seen that "call function, assign output, test output using semicolon, and do all this on the same line as the if-conditional" pattern before, but looking over the rest of the code and also some go idiom paged it seems like it's a common pattern?  i'm happy to continue using this style since it's convention!

Reply:
One additional note while i'm here - Danny and i decided to refactor the function to have an early return on the type being not GA, which un-nests a big block of code and might be a bit easier to follow.  Curious for your thoughts there!  It's semantically equivalent, but we both prefer early returns and thought it was worthwhile to do for readability.

Reply:
I am also a fan of early returns. Sounds good! I'll take a look at the changed code.